File: NEC_Challenge.Core/Models/CryptoCurrencyConversion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NEC_Challenge.Core.Models
{
    public class CryptoCurrencyConversion
    {
        public CryptoCurrencyConversion()
        {
            Conversions = new List<Conversion>();
        }

        public CryptoCurrency CurrencyToConvert { get; set; }
        public List<Conversion> Conversions { get; set; }
    }
}

File: NEC_Challenge.Web/src/app/core/models/crypto-currency-conversion.ts
import { Conversion } from "./conversion";
import { CryptoCurrency } from "./crypto-currency";

export class CryptoCurrencyConversion {
    currencyToConvert: CryptoCurrency;
    conversions: Array<Conversion>;

    constructor(_currencyToConvert: CryptoCurrency, _conversions: Array<Conversion>) 
        {
        this.currencyToConvert = _currencyToConvert;
        this.conversions = _conversions;
    }

    public static adapt(item: any): CryptoCurrencyConversion {
        return new CryptoCurrencyConversion(
            item.currencyToConvert,
            item.conversions
        );
    }
}